internal/stores/hostdb.go

```diff
@@ -660,20 +660,50 @@ func (ss *SQLStore) ProcessConsensusChange(cc modules.ConsensusChange) {
 				announcement: ha,
 			})
 		})
+		// Update RevisionHeight and RevisionNumber for our contracts.
+		for _, txn := range sb.Transactions {
+			for _, rev := range txn.FileContractRevisions {
+				if _, isOurs := ss.isOurContract[types.FileContractID(rev.ParentID)]; isOurs {
+					ss.unappliedRevisions[types.FileContractID(rev.ParentID)] = revisionUpdate{
+						height: height,
+						number: rev.NewRevisionNumber,
+					}
+				}
+			}
+			// Get ProofHeight for our contracts.
+			for _, sp := range txn.StorageProofs {
+				if _, isOurs := ss.isOurContract[types.FileContractID(sp.ParentID)]; isOurs {
+					ss.unappliedProofs[types.FileContractID(sp.ParentID)] = height
+				}
+			}
+		}
 		height++
 	}
 
 	ss.unappliedAnnouncements = append(ss.unappliedAnnouncements, newAnnouncements...)
 	ss.unappliedCCID = cc.ID
 
-	// Apply new announcements
-	if time.Since(ss.lastAnnouncementSave) > ss.persistInterval || len(ss.unappliedAnnouncements) >= announcementBatchSoftLimit {
+	// Apply updates.
+	if time.Since(ss.lastAnnouncementSave) > ss.persistInterval ||
+		len(ss.unappliedAnnouncements) >= announcementBatchSoftLimit ||
+		len(ss.unappliedRevisions) > 0 || len(ss.unappliedProofs) > 0 {
```

Comment:
`nit` should we have every condition on its own line?

Reply:
I thought it looked more readable.